x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/pipeline_form.tsx

```diff
@@ -107,10 +126,42 @@ export const PipelineForm: React.FunctionComponent<PipelineFormProps> = ({
   );
 
   const onProcessorsChangeHandler = useCallback<OnUpdateHandler>(
-    (arg) => (processorStateRef.current = arg),
-    []
+    (arg) => {
+      processorStateRef.current = arg;
+
+      const currentProcessorsState = processorStateRef.current?.getData();
+
+      // Calculate if the current processor state has changed compared to the
+      // initial processors state.
+      setAreProcessorsDirty(
+        !isEqual(
```

Comment:
Interesting, I havent thought about that! It seems that just hitting the save button ends up with the same state but some attributes are in different positions so it turns out that the check to see if both states are equal returns false. It makes it all quite complicated given that the processors state is tracked outside of the form itself, otherwise this would be a more trivial operation to perform 🤔 

<img width="1246" alt="Screenshot 2024-06-06 at 14 45 42" src="https://github.com/elastic/kibana/assets/1191206/28da47eb-898c-4433-ba94-e2384b51b30c">

Reply:
I've added a [new commit](https://github.com/elastic/kibana/pull/183699/commits/6038780dc2156cf0b00b072fd7ed8c62110b0cad) that ignores undefined values when performing the deeply comparisson check to know if the processors are dirty or not.

Reply:
Great, thanks @sabarasaba! Tested locally and now the dirty processors seem to be detected correctly.

On a side note, I noticed that since the last commit, the modal text and color have been changed. Was this intentional? Also, when you try to close the modal, it navigates out of the page instead of staying on the page to keep editing, which looks counter-intuitive to me:

https://github.com/elastic/kibana/assets/59341489/3a51da13-3f0c-4613-b15b-ec54053a6938

Reply:
Yeah we updated the design and copy of the confirmation modal to be according to the design guideliness. Good catch on the X, doesnt seem that core gives us any mechanism to detect what was clicked besides the main CTA, which makes inverting the actions (as in the previous commit) like in the design guideline impossible to achieve. I've returned those changes back, and will create an issue for core and one for us to update this after.